### ISSUE TITLE
[Feature] Sync with repo

### ISSUE THREAD

**Issue by lukasoppermann** [figma/code-snippet-editor-plugin#22]
Hey, it would be really cool to allow for improved syncronization with a repository. 

I imagine a way to login to a github repo from the plugin and link to a specific `.json` file with the structure in it. Whenever the file changes, the plugin could show an update button or something to pull in the latest changes.

## Why?
A common issue is that docs, code playgrounds and figma get out of sync. Automating the update process and putting the source of truth onto github could help a lot.

**Comment by jake-figma** [comment]
I agree on this being a great addition. This would also unblock desires for additional "global" node-type based templates. see #21. Currently thinking over a few different approaches here. One being OAuth'd as you describe, also considering a lighter lift public JSON endpoint sync as a good first step in that direction.

Additionally, our team is thinking about something like this at a feature level, so the request itself is great feedback I'm sending that way.
